pennylane/templates/embeddings/basis.py

```diff
@@ -12,72 +12,62 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 r"""
-Contains the ``BasisEmbedding`` template.
+Contains the BasisEmbedding template.
 """
 # pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
 import pennylane as qml
-from pennylane.templates.decorator import template
+from pennylane.operation import Operation, AnyWires
 from pennylane.wires import Wires
 
 
-def _preprocess(features, wires):
-    """Validate and pre-process inputs as follows:
-
-    * Check that the features tensor is one-dimensional.
-    * Check that the first dimension of the features tensor
-      has length :math:`n`, where :math:`n` is the number of qubits.
-    * Check that the entries of the features tensor are zeros and ones.
+class BasisEmbedding(Operation):
+    r"""Encodes :math:`n` binary features into a basis state of :math:`n` qubits.
 
-    Args:
-        features (tensor_like): input features to pre-process
-        wires (Wires): wires that template acts on
+    For example, for ``features=np.array([0, 1, 0])``, the quantum system will be
+    prepared in state :math:`|010 \rangle`.
 
-    Returns:
-        array: numpy array representation of the features tensor
-    """
-    shape = qml.math.shape(features)
+    .. warning::
 
-    if len(shape) != 1:
-        raise ValueError(f"Features must be one-dimensional; got shape {shape}.")
+        ``BasisEmbedding`` calls a circuit whose architecture depends on the binary features.
+        The ``features`` argument is therefore not differentiable when using the template, and
+        gradients with respect to the argument cannot be computed by PennyLane.
 
-    n_features = shape[0]
-    if n_features != len(wires):
-        raise ValueError(f"Features must be of length {len(wires)}; got length {n_features}.")
+    Args:
+        features (tensor-like or Iterable): binary input of shape ``(n, )``
```

Comment:
Here we also allow lists as inputs, since the template is not differentiable, and lists are a natural representation of bitstrings.